api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -32,6 +32,27 @@
 import org.apache.iceberg.exceptions.RuntimeIOException;
 
 public interface CloseableIterable<T> extends Iterable<T>, Closeable {
+
+  default CloseableIterator<T> closeableIterator() {
+    Iterator<T> iterator = iterator();
+    return new CloseableIterator<T>() {
+      @Override
+      public void close() throws IOException {
+        CloseableIterable.this.close();
```

Comment:
This isn't correct. `CloseableIterable#close` will close all open iterators created by the `Iterable`. The purpose is to enable try-with-resources instead of relying on callers to remember to close:

```java
try (CloseableIterable<T> iterable = open(...)) {
  for (T item : iterable) {
    process(item);
  }
}
```

If an iterator closed the iterable, then it would close other iterators that may be still used in other threads.

Reply:
Thanks for pointing this out, I've fixed this in the new patch.